src/extras/render-passes/render-pass-depth-aware-blur.js

```diff
@@ -8,18 +11,87 @@ import { shaderChunks } from '../../scene/shader-lib/chunks/chunks.js';
  * @ignore
  */
 class RenderPassDepthAwareBlur extends RenderPassShaderQuad {
-    constructor(device, sourceTexture) {
-        super(device);
-        this.sourceTexture = sourceTexture;
-
-        this.shader = this.createQuadShader('DepthAwareBlurShader', shaderChunks.screenDepthPS + /* glsl */`
+    /**
+     * Initializes a new blur render pass. It has to be called before the render pass can be used.
+     *
+     * @param {import('../../platform/graphics/render-target.js').RenderTarget|null} [renderTarget] - The render
+     * target to render into (output).
+     * @param {Object} options - Options for configuring the blur effect.
+     * @param {import('../../platform/graphics/texture.js').Texture} [options.resizeSource] - The source texture to be blurred.
+     * @param {number} [options.scaleX] - The scale factor for the render target width. Defaults to 1.
+     * @param {number} [options.scaleY] - The scale factor for the render target height. Defaults to 1.
+     * @param {number} [options.kernelSize] - The size of the blur kernel. Defaults to 7.
+     * @param {boolean} [options.depthAware] - Whether the blur should be depth-aware. Defaults to false.
+     * @param {BLUR_GAUSSIAN|import('../../scene/constants.js').BLUR_BOX} [options.type] - The type of blur to apply. Defaults to BLUR_GAUSSIAN.
+     * @param {Vec2} [options.direction] - The direction of the blur. Defaults to (1, 0).
+     * @param {string} [options.channels] - The color channels to apply the blur to ('r'|'g'|'b'|'a'|'rg'|..|'ba'|'rgb'|'gba'|'rgba'). Defaults to 'rgba'.
+     */
+    init(renderTarget = null, options = {}) {
```

Comment:
I would not override the `init` function, but instead add a `setup` function which does all this.